### ISSUE TITLE
pubsub: lack of create/destroy functions makes testing manual and tedious

### ISSUE THREAD

**Issue by ijt** [google/go-cloud#1091]
I keep having to manually purge an SNS queue for some reason while running the new `awspubsub` tests in record mode, otherwise there is a test failure like this:

```
[ ~/gocloud.dev/pubsub/awspubsub ] go test -record 
--- FAIL: TestConformance (10.56s)
    --- FAIL: TestConformance/TestSendReceiveTwo (3.99s)
        replay.go:69: Recording into golden file testdata/TestConformance/TestSendReceiveTwo
        drivertest.go:233: sub #1: Sort({[]*pubsub.Message})[1->?]:
            	-: &pubsub.Message{Body: []uint8{0x30}, Metadata: map[string]string{"a": "0"}, ack: (func())(0x014c69a0), isAcked: true}
            	+: <non-existent>
            Sort({[]*pubsub.Message})[?->2]:
            	-: <non-existent>
            	+: &pubsub.Message{Body: []uint8{0x32}, Metadata: map[string]string{"a": "2"}}
FAIL
exit status 1
FAIL	gocloud.dev/pubsub/awspubsub	10.717s
```

If we had create/destroy functions for subscriptions (and topics for consistency) then it would allow for more automated and reliable tests that don't require manual maintenance of cloud resources.

**Comment by jba** [comment]
Yes, I was seeing that when I was recording the gcppubsub tests the other day. I manually drained with `gcloud`.

The advantage of draining over creating is that when your test crashes, you don't leave around a cloud resource that needs to be managed. A disadvantage is that it's flakier; you have to wait a bit to make sure you really reda all the messages.

In either case, the harness could provide the necessary functions. No need to add them to the API.

**Comment by vangent** [comment]
> the harness could provide the necessary functions
+1. That's what `runtimevar` does -- the harness has functions to create, modify, and destroy variables, but those functions are not part of the public `runtimevar` API.

It wouldn't even necessarily have to be part of the harness; e.g., `harness.MakeTopic` could do the drain.

**Comment by vangent** [comment]
> part of the harness
To clarify, I meant that it wouldn't need to be a new function on the harness that the conformance test explicitly called.

**Comment by ijt** [comment]
Fixed by #1107
